File: 프로그래머스/lv1/12934. 정수 제곱근 판별/정수 제곱근 판별.java
class Solution {
    public long solution(long n) {
    		  long answer = 0;
    	  
    	  long i= 1;
    	  long a = 0;
    	  while(true) {
    		  if(Math.pow( i, 2) >=n) {
    			  a = i;
    			  break; 
    		  }
    		  i++;
    		  
    	  }
    	  
    	  if(Math.pow(a, 2) == n) {
    		  answer= (long) Math.pow(a+1, 2);
    	  }
    	  
    	  if(answer==0) {
    		  answer=-1;
    	  }
          return answer;
    	  
    }
}

File: 프로그래머스/1/12934. 정수 제곱근 판별/정수 제곱근 판별.js
function solution(n) {
    var answer = 0;
    var sqrt = Math.round( Math.sqrt(parseInt(n)) , 2) ;
    
    if( Math.pow(sqrt,2) == n ){
        return Math.pow( ( sqrt+1) , 2 );
    } else{
        return -1 ; 
    }
    
    return answer;
}